services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/task.py

```diff
@@ -139,6 +141,34 @@ async def add_service(self, scheduler_data: SchedulerData) -> None:
             self._enqueue_observation_from_service_name(scheduler_data.service_name)
             logger.debug("Added service '%s' to observe", scheduler_data.service_name)
 
+    def list_services(
+        self,
+        *,
+        user_id: Optional[UserID] = None,
+        project_id: Optional[ProjectID] = None,
+    ) -> list[NodeID]:
+        """Returns the list of tracked service UUIDs"""
+        all_tracked_service_uuids = list(self._inverse_search_mapping)
+        if user_id == project_id == None:
+            return all_tracked_service_uuids
+        # let's filter
+        def _filter_scheduler_data(node_id: NodeID) -> bool:
```

Comment:
MINOR:  too many "filter" prefixes ... My proposal

```python
filtered_tracked_service_uuids = filter( _is_scheduled, all_tracked_service_uuids)
```

there is no need for this generator, right? ``(n for n in all_tracked_service_uuids)``